packages/framework/fluid-framework/legacy.d.ts

```diff
@@ -0,0 +1,18 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/*
+ * THIS FILE SHOULD NOT BE DELETED.
+ *
+ * This file is needed for compatibility with TypeScript's Node10 module resolution. In most other packages it is
+ * auto-generated by `flub generate entrypoints` in @fluid-tools/build-cli, and is thus gitignored in most packages.
+ *
+ * However, this package is unique in that it re-exports some legacy exports from other packages directly in a source
+ * file legacy.ts, and does not need to use API trimming or `flub generate entrypoints` to create the trimmed API rollup
+ * files. Since `flub generate entrypoints` also creates the Node10-compatible type stubs, this file is committed
```

Comment:
@jason-ha Claimed we don't support node10 module resolution when he wrote https://github.com/microsoft/FluidFramework/pull/20466/files#diff-b200e6c6f6b045be32bbbea01490bfc0fc1b75782caa3efab816768fda2ee232R27 for me. Do we actually intend to support it?

Reply:
Officially, end users should switch to Node16 and Bundler ASAP.  End of story.

...but to ease certain partner(s) with the transition, we've temporarily added the stubs necessary for TypeScript transpilation to succeed with '"moduleResolution": "node"' so they don't get hit with a double-whammy of needing to bite off Node16 and move to the '/legacy' entry point in one step.

However, we still rely on a modern Node environment or bundler that understands our export maps in the package.json, so it's not true node10 support.

Reply:
Thats a great explanation: might be nice ot put it in the comment!